core/state_processor_rip7560.go

```diff
@@ -346,12 +344,32 @@ func ApplyRip7560ExecutionPhase(config *params.ChainConfig, vpr *ValidationPhase
 	accountExecutionMsg := prepareAccountExecutionMessage(vpr.Tx, evm.ChainConfig())
 	executionResult, err := ApplyMessage(evm, accountExecutionMsg, gp)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	var paymasterPostOpResult *ExecutionResult
 	if len(vpr.PaymasterContext) != 0 {
 		paymasterPostOpResult, err = applyPaymasterPostOpFrame(vpr, executionResult, evm, gp, statedb, header)
 	}
+	return executionResult, paymasterPostOpResult, err
+}
+
+func ApplyRip7560ExecutionPhase(config *params.ChainConfig, vpr *ValidationPhaseResult, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, cfg vm.Config) (*types.Receipt, error) {
+
+	// First we need to apply execution and postop messages to a copy, to be able to revert if
+	statedbCopy := statedb.Copy()
```

Comment:
why copy and not [save/revert](https://github.com/eth-infinitism/go-ethereum/blob/RIP-7560-revision-2/core/vm/evm.go#L291-L306) pattern?
it has the benefit of not copying data all the time, only safely restore in case of revert.

Reply:
The problem when using `stateDb.Snapshot()... stateDb.RevertToSnapshot()` is that it is not meant to be used after two `ApplyMessage()` calls it seems - it simply crashes when I do it. That's why I did it that way. Maybe there's a better way.